### PR TITLE
checker: disallow reserve type name for const

### PR DESCRIPTION
**Pull request by Delta456**
FIxes https://github.com/vlang/v/issues/16582

### CODE REVIEW

File: vlib/orm/orm.v [hunk 1]
```diff
@@ -18,7 +18,6 @@ pub const (
 		ast.f32_type_idx,
 		ast.f64_type_idx,
 	]
-	string   = ast.string_type_idx
```
Comment (Delta456) [inline vlib/orm/orm.v:21]: I am not sure why is this needed for 😕🤔

**Review by spytheman — APPROVED** [review]
